src/notifications.c

```diff
@@ -330,6 +336,81 @@ void ShardingEvent(RedisModuleCtx *ctx, RedisModuleEvent eid, uint64_t subevent,
   }
 }
 
+static bool in_asm_trim = false;
+static bool in_asm_import = false;
+
+void ClusterSlotMigrationEvent(RedisModuleCtx *ctx, RedisModuleEvent eid, uint64_t subevent, void *data) {
+  REDISMODULE_NOT_USED(eid);
+  REDISMODULE_NOT_USED(data);
+
+  switch (subevent) {
+
+    case REDISMODULE_SUBEVENT_CLUSTER_SLOT_MIGRATION_IMPORT_STARTED:
+      RedisModule_Log(RSDummyContext, "notice", "Got ASM import started event.");
+      in_asm_import = true;
+      should_filter_slots = true;
+      workersThreadPool_OnEventStart();
+      break;
+    case REDISMODULE_SUBEVENT_CLUSTER_SLOT_MIGRATION_IMPORT_FAILED:
+    case REDISMODULE_SUBEVENT_CLUSTER_SLOT_MIGRATION_IMPORT_COMPLETED:
+      RedisModule_Log(RSDummyContext, "notice", "Got ASM import %s event.", subevent == REDISMODULE_SUBEVENT_CLUSTER_SLOT_MIGRATION_IMPORT_FAILED ? "failed" : "completed");
+      in_asm_import = false;
+      should_filter_slots = in_asm_trim;
+      // Since importing is done in a part-time job while redis is running other commands, we notify
+      // the thread pool to no longer receive new jobs, and terminate the threads ONCE ALL PENDING JOBS ARE DONE.
+      workersThreadPool_OnEventEnd(false);
+      Slots_DropCachedLocalSlots(); // Local slots have changed, drop the cache

```

Comment:
Config doesn't change on FAILED event. If I understand your logic correctly, you don't need to drop it, fyi.